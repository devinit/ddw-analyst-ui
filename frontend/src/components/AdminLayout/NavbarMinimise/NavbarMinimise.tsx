import * as React from 'react';

export const NavbarMinimise: React.SFC = () => {
  return (
    <div className="navbar-minimize">
      <button id="minimizeSidebar" className="btn btn-just-icon btn-white btn-fab btn-round">
        <i className="material-icons text_align-center visible-on-sidebar-regular">more_vert</i>
        <i className="material-icons design_bullet-list-67 visible-on-sidebar-mini">view_list</i>
      </button>
    </div>
  );
};
