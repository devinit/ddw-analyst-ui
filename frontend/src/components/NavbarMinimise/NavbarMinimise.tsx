import * as React from 'react';

export class NavbarMinimise extends React.Component {
  render() {
    return (
      <div className="navbar-minimize">
        <button className="btn btn-just-icon btn-white btn-fab btn-round" onClick={ this.onClick }>
          <i className="material-icons text_align-center visible-on-sidebar-regular">more_vert</i>
          <i className="material-icons design_bullet-list-67 visible-on-sidebar-mini">view_list</i>
        </button>
      </div>
    );
  }

  private onClick() {
    if (document.body.className.indexOf('sidebar-mini') > -1) {
      document.body.classList.remove('sidebar-mini');
    } else {
      document.body.classList.add('sidebar-mini');
    }
  }
}
