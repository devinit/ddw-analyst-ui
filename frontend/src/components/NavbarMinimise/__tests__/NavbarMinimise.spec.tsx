import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { NavbarMinimise } from '../NavbarMinimise';

test('renders correctly', () => {
  const renderer = TestRenderer
    .create(<NavbarMinimise/>)
    .toJSON();

  expect(renderer).toMatchSnapshot();
});
