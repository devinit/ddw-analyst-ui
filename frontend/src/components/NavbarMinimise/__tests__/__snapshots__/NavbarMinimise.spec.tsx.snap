// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar-minimize"
>
  <button
    className="btn btn-just-icon btn-white btn-fab btn-round"
    id="minimizeSidebar"
  >
    <i
      className="material-icons text_align-center visible-on-sidebar-regular"
    >
      more_vert
    </i>
    <i
      className="material-icons design_bullet-list-67 visible-on-sidebar-mini"
    >
      view_list
    </i>
  </button>
</div>
`;
